chimney-macro-commons/src/main/scala-3/io/scalaland/chimney/internal/compiletime/ExprPromisesPlatform.scala

```diff
@@ -19,7 +19,22 @@ private[compiletime] trait ExprPromisesPlatform extends ExprPromises { this: Def
         case NameGenerationStrategy.FromExpr(expr)  => freshTermName[From](expr, usageHint)
       }
 
-    protected def createRefToName[From: Type](name: ExprPromiseName): Expr[From] = Ref(name).asExprOf[From]
+    protected def createRefToName[From: Type](name: ExprPromiseName): Expr[From] = {
+      val From = TypeRepr.of[From]
+
+      val fixedRef = for {
+        sym <- Option(From.typeSymbol)
+        if sym.flags.is(Flags.Enum | Flags.StableRealizable) && !sym.flags.is(Flags.JavaStatic)
+        parentSymbol <- From.baseClasses.find(_.flags.is(Flags.Enum | Flags.Sealed))
+        parentType = From.baseType(parentSymbol)
+        fixedFrom = Type.platformSpecific.fromUntyped(parentType.memberType(sym)).as_??
+      } yield {
+        import fixedFrom.Underlying as FixedFrom
+        Ref(name).asExprOf[FixedFrom].asInstanceOf[Expr[From]]
+      }
```

Comment:
Naive attempt to apply `TypeRepr.of[A].memberType(subtype)` fix once we already matched types. Does not work.